File: Difficulty: Medium/Root to Leaf Paths/root-to-leaf-paths.cpp
/*

Definition for Binary Tree Node
struct Node
{
    int data;
    struct Node* left;
    struct Node* right;

    Node(int x){
        data = x;
        left = right = NULL;
    }
};
*/

class Solution {
  public:
    void pospaths(Node* node,vector<vector<int>> &ans,vector<int> &pa){
        if (node == NULL) {
            return;
        }
        
        pa.push_back(node->data);
        if(node->left==NULL && node->right==NULL){
            ans.push_back(pa);
        }
        else{
            pospaths(node->left,ans,pa);
            pospaths(node->right,ans,pa);
        }
        
        pa.pop_back();
    }
    
    vector<vector<int>> Paths(Node* root) {
        // code here
        vector<vector<int>> ans;
        vector<int>pa;
        if(root==NULL) return ans;
        pospaths(root,ans,pa);
        return ans;
    }
};

File: Difficulty: Medium/Root to Leaf Paths/root-to-leaf-paths.py
from typing import Optional
from collections import deque

from typing import List

"""

definition of binary tree node.
class Node:
    def _init_(self,val):
        self.data = val
        self.left = None
        self.right = None
"""

class Solution:
    def getpath(self,root,curr_path,ans):
        if root is None:
            return 
        curr_path.append(root.data)
        if root.left==None and root.right==None:
            ans.append(list(curr_path))
        else:
            self.getpath(root.left,curr_path,ans)
            self.getpath(root.right,curr_path,ans)
        curr_path.pop()
        
    
    def Paths(self, root):
        # code here
        ans=[]
        if root is None:
            return ans
        curr_path=[]
        self.getpath(root,curr_path,ans)
        return ans
        